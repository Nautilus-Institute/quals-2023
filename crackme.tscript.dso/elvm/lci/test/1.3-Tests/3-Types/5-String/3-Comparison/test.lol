HAI 1.3
    CAN HAS STRING?

    I HAS A string ITZ "::hello"

    I HAS A char ITZ I IZ STRING'Z AT YR string AN YR 0 MKAY
    VISIBLE char AN " - " AN "::"

    BOTH SAEM char AN "::", O RLY?
    YA RLY
        VISIBLE "success"
    NO WAI
        VISIBLE "fail"
    OIC

    VISIBLE char AN " - " AN ":(3A)"
    BOTH SAEM char AN ":(3A)", O RLY?
    YA RLY
        VISIBLE "success"
    NO WAI
        VISIBLE "fail"
    OIC

    I HAS A char2 ITZ I IZ STRING'Z AT YR string AN YR 1 MKAY
    VISIBLE char2 AN " - " AN "h"
    BOTH SAEM char2 AN "h", O RLY?
    YA RLY
        VISIBLE "success"
    NO WAI
        VISIBLE "fail"
    OIC

    I HAS A char3 ITZ "::"
    VISIBLE char3 AN " - " AN "::"
    BOTH SAEM char3 AN "::", O RLY?
    YA RLY
        VISIBLE "success"
    NO WAI
        VISIBLE "fail"
    OIC
KTHXBYE
