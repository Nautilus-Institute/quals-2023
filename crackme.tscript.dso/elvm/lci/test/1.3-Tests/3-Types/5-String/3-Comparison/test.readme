This test checks that strings containing escape sequences are compared
correctly.
