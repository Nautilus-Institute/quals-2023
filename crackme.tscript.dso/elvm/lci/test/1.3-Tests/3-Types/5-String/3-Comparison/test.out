: - :
success
: - :
success
h - h
success
: - :
success
